### PR TITLE
Update ruby to ruby 3.1

### PR DESCRIPTION
**Pull request by vpereira**
Update ruby to ruby 3.1

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -4,21 +4,23 @@ ARG USER_ID=1000
 ARG GROUP_ID=100
 ARG USER_NAME=foo
 
-RUN zypper -n install openssh-clients ansible ruby ruby-devel bash git libxml2-devel gcc make shadow vim curl
+RUN zypper ar -f -G https://download.opensuse.org/repositories/devel:/languages:/ruby/openSUSE_Leap_15.3/devel:languages:ruby.repo
+RUN zypper --gpg-auto-import-keys refresh
+
+RUN zypper -n install openssh-clients ansible ruby3.1 ruby3.1-devel bash git libxml2-devel gcc make shadow vim curl libxslt-devel
```
Comment (hennevogel) [inline Dockerfile:10]: *nitpick* `ruby3.1-devel` pulls in `ruby3.1`...

File: Dockerfile [hunk 2]
```diff
@@ -4,21 +4,23 @@ ARG USER_ID=1000
...
 RUN zypper -q clean -a
 
 RUN useradd -l -u ${USER_ID} -g ${GROUP_ID} --home-dir /home/${USER_NAME} --create-home ${USER_NAME}
 
-RUN gem install bundler -v '~> 2.2'
+RUN gem install bundler
```
Comment (hennevogel) [inline Dockerfile:15]: `ruby3.1` doesn't bring `gem` in $PATH. Only `gem.ruby3.1`.

Comment (hennevogel) [reply Dockerfile:15]: But `ruby3.1` brings `bundler`!

```
frontend@6c5cefb960bf:/obs/src/api> /usr/bin/bundler -v
Bundler version 2.3.7
frontend@6c5cefb960bf:/obs/src/api> rpm -qf /usr/bin/bundler
ruby3.1-3.1.1-13.2.x86_64
```

**Review by hennevogel — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 1 above)
Comment (vpereira) [reply Dockerfile:10]: yes, I know, I however always pull both to make sure to be explicit here. I can change however without problem :+1:

File: Dockerfile (hunk 2 above)
Comment (vpereira) [reply Dockerfile]: updated, please take a look @hennevogel

**Pull request merged**